### PR TITLE
DOC: fix levels in user/explain/figure

### PR DESCRIPTION
**Pull request by jklymak**
## PR summary

This fixes user/explain/figure, user/explain/axes, user/explain/artists to have a more conventional TOC page, followed by all the sub-pages.  This is more like what sphinx expects, and makes the relevant sections show up as peers in the TOC.

Note there are no content changes other than the new TOC pages.  

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

### CODE REVIEW

**Comment by jklymak** [comment]
As an aside, learned that you can do 

```
.. toctree::
    :maxdepth: 2

    axes_intro

.. toctree::
    :maxdepth: 1

    arranging_axes
    colorbar_placement
```

And the TOC in the sidebar will show all of the entries, but the page will render with a nested TOC with just the first entry expanded.

**Comment by jklymak** [comment]
BTW, @story645 I'm all for having these TOCs be more visual, perhaps even with gallery-like cards.  But I think these fixes should go in first just to get it organized better.

**Comment by rcomer** [comment]
I like the the change to the TOCs, but I feel like something is now missing from the [top level figure page](https://output.circle-artifacts.com/output/job/0a192c54-7d64-4063-b550-01c88d12ba21/artifacts/0/doc/build/html/users/explain/figure/index.html).  Maybe just a very basic description of what a figure is for between the title and the code block?

**Comment by jklymak** [comment]
Ooops, yes, I meant to do that.  

I also made the users_explain/index.rst into cards and maxdepth: 2 for better skim-ability.  Some of the cards will need rebalancing later.

**Comment by jklymak** [comment]
https://output.circle-artifacts.com/output/job/bd5db14f-19b4-4a3d-9fdb-0884aca17cd4/artifacts/0/doc/build/html/users/explain/index.html

File: galleries/users_explain/figure/index.rst [hunk 1]
```diff
@@ -1,19 +1,6 @@
...
-                     layout='constrained')
-    fig.suptitle('Figure')
-    ax = fig.add_subplot()
-    ax.set_title('Axes', loc='left', fontstyle='oblique', fontsize='medium')
+++++++++++++++++++++
+Figures and backends
+++++++++++++++++++++
 
 When looking at Matplotlib visualization, you are almost always looking at
 Artists placed on a `~.Figure`.  In the example above, the figure is the
```
Comment (rcomer) [inline galleries/users_explain/figure/index.rst:6]:
```suggestion
Artists placed on a `~.Figure`.  In the example below, the figure is the
```

**Comment by jklymak** [comment]
Yeah, this only moved things, so if you want changes to the actual content, happy to look at PRs that do that. 

The catchall is inelegant, but the whole top index needs work.  The main point here is fixing the users/explain/*/index etc so that they are just indexes and the sub info are peers instead of the awkward.   I did have a version where each item was its own card, but that is really worse as they're a bunch of very empty cards.  I think we can follow up on that and that the proposed structure here is better than what we have.  If you'd prefer I can revert the user/explain/index changes, and make a new PR to mess around with that.

**Comment by jklymak** [comment]
I think I did mix and match. The subjects with substantive number of sub-headings have a separate card.  Then there are the catch-alls.

**Comment by jklymak** [comment]
Sure agreed that definitely needs to be fixed. But that is orthogonal to this pr.

File: galleries/users_explain/index.rst [hunk 2]
```diff
@@ -6,16 +6,57 @@
...
+        :padding: 2
+
+        .. toctree::
+            :maxdepth: 2
+
+            axis/index
+            colors/index
+            text/index
+            animations/index
+            toolkits/index
```
[the remark this answers is not in the document]
Comment (jklymak) [reply galleries/users_explain/index.rst]: I've reverted the changes to `users_explain/index.rst` so that can be in a separate PR

**Comment by jklymak** [comment]
- I've reverted the change to `users/explain/index`.
- this PR largely only moved material, it didn't edit anything in the moved files.  If there are changes desired to the moved material, those should be their own PR or issue, and not part of reviewing this PR.  
- the PR has new intro paragraphs for explain/figure/index, explain/axes/index, and explain/artist/index and a figure in each of those. These are copied almost verbatim from explain/*/*_intro.rst.

**Review by rcomer — APPROVED** [review]
I’ve not been through in detail, only clicked around the built docs.  I think they look great!  Since @story645 has been through the details and already approved, I think this is good to go.

@jklymak do you want to squash the commits?  Either way feel free to self merge.

**Pull request merged**